openfl/utilities/optimizers/torch/fedprox.py

```
@@ -115,7 +116,11 @@ def step(self, closure=None):
             nesterov = group["nesterov"]
             mu = group["mu"]
             w_old = group["w_old"]
-            for p, w_old_p in zip(group["params"], w_old):
+            
+            # Skip FedProx regularization if w_old is not set or mu is 0
+            apply_proximal = w_old is not None and mu > 0
```

IMO, `FedProx` should always attempt to apply the proximal term. 

If a user sets `mu==0` regardless of `w_old`, it'll mathematically cancel out the proximal term anyway (`mu*(p-w_old_p)`). 

`w_old is not None and mu > 0` should be the expected default condition if a user is trying to use FedProx

If `w_old is None and mu > 0`, we should raise an error or warning that the user needs to set the old weights prior to training. This condition indicates that a user wanted to use FedProx so just omitting the proximal term if the user forgets to set the old weights could lead to a silent bug for them. 

We also probably shouldn't let m < 0, though technically it isn't mathematically wrong, it's just counterintuitive since it'll reward deviations (so, maybe a warning)

---

I have added requested conditions to fedprox.py.